aws/resource_aws_codeartifact_repository.go

```diff
@@ -146,6 +162,37 @@ func resourceAwsCodeArtifactRepositoryUpdate(d *schema.ResourceData, meta interf
 		return fmt.Errorf("error updating CodeArtifact Repository: %w", err)
 	}
 
+	if d.HasChange("external_connections") {
+		if v, ok := d.GetOk("external_connections"); ok {
```

Comment:
Need to handle multiple external connections?
Become easier if we change to set as set difference can be used to determine additions and deletions.

Reply:
See comment above. Let's have `MaxItems: 1,` in the schema.

Reply:
added